tools/crdcheck/main.go

```diff
@@ -0,0 +1,82 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2020-2021 Authors of Cilium
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+
+	crdv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+)
+
+const mandatoryCategory = "cilium"
+
+type checkCRDFunc func(*crdv1.CustomResourceDefinition) error
+
+var allChecks = []checkCRDFunc{
+	checkForCategory,
+}
+
+func main() {
+	if len(os.Args) != 2 {
+		log.Fatal(fmt.Sprintf("usage: %s <path>", os.Args[0]))
+	}
+
+	_ = crdv1.AddToScheme(scheme.Scheme)
+
+	if err := filepath.Walk(os.Args[1], func(path string, info os.FileInfo, _ error) error {
+		if info.IsDir() {
+			return nil
+		}
+
+		if ext := filepath.Ext(path); ext != ".yaml" && ext != ".yml" {
+			return nil
+		}
+
+		fileContent, err := ioutil.ReadFile(path)
+		if err != nil {
+			return err
+		}
+
+		obj, _, err := scheme.Codecs.UniversalDeserializer().Decode(fileContent, nil, nil)
+		if err != nil {
+			return err
+		}
+
+		crd, ok := obj.(*crdv1.CustomResourceDefinition)
+		if !ok {
+			return nil
+		}
+
+		for _, f := range allChecks {
+			if err = f(crd); err != nil {
```

Comment:
currently, we have only one check, so i didn't use multi error here. Let's see how it's going.